File: lua/pangram/pangram.lua
local alphabet = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l',
  'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'}

local function count_letters(sentence)
  local counts = {}
  for letter in string.gmatch(string.lower(sentence), "[a-z]") do
    counts[letter] = (counts[letter] or 0) + 1
  end
  return counts
end

local function contains_all(counts, alphabet)
  for _, letter in pairs(alphabet) do
    if not counts[letter] then return false end
  end
  return true
end

return function(sentence)
  local counts = count_letters(sentence)
  return contains_all(counts, alphabet)
end

File: elixir/pangram/pangram.exs
defmodule Pangram do
  @doc """
  Determines if a word or sentence is a pangram.
  A pangram is a sentence using every letter of the alphabet at least once.

  Returns a boolean.

    ## Examples

      iex> Pangram.pangram?("the quick brown fox jumps over the lazy dog")
      true

  """
  @spec pangram?(String.t) :: boolean
  def pangram?(sentence) do
    count_unique_letters(sentence) == 26
  end

  defp alphabet?(c), do: ?a <= c && ?z >= c

  defp count_unique_letters(sentence) do
    String.downcase(sentence)
    |> String.to_charlist
    |> Enum.filter(&alphabet?/1)
    |> Enum.into(MapSet.new)
    |> MapSet.size
  end
end

# Enum.into(%{a: 1}, %{b: 2})